File: src/main/java/ru/hubsmc/hubscore/module/donate/HubsDonate.java
package ru.hubsmc.hubscore.module.donate;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import ru.hubsmc.hubscore.CoreModule;
import ru.hubsmc.hubscore.module.donate.commands.CartCommand;
import ru.hubsmc.hubscore.module.donate.commands.DonateCommand;

import java.util.List;

import static ru.hubsmc.hubscore.PluginUtils.setCommandExecutorAndTabCompleter;

public class HubsDonate extends CoreModule {

    @Override
    public boolean onEnable() {
        setCommandExecutorAndTabCompleter(new DonateCommand());
        setCommandExecutorAndTabCompleter(new CartCommand());
        return true;
    }

    @Override
    public void onDisable() {
    }

    @Override
    public void onReload() {
    }

    @Override
    public void onPlayerJoin(Player player) {
    }

    @Override
    public void onPlayerLeave(Player player) {
    }

    @Override
    public void onSchedule(byte min) {
    }

    @Override
    public boolean onCommandExecute(CommandSender sender, Command command, String label, String[] args) {
        return false;
    }

    @Override
    public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) {
        return null;
    }

    public static void buyDonate(Player player, String donateKey) {
    }

}

File: src/main/java/su/hubs/hubscore/module/donate/HubsDonate.kt
package su.hubs.hubscore.module.donate

import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import su.hubs.hubscore.CoreModule
import su.hubs.hubscore.PluginUtils
import su.hubs.hubscore.module.chesterton.internal.item.ChestertonItem
import su.hubs.hubscore.module.chesterton.internal.menu.ChestertonMenu
import su.hubs.hubscore.module.chesterton.internal.menu.ConfirmMenu
import su.hubs.hubscore.module.donate.commands.CartCommand
import su.hubs.hubscore.module.donate.commands.DonateCommand
import su.hubs.hubscore.module.essentials.HubsEssentials
import su.hubs.hubscore.module.values.ValueType
import su.hubs.hubscore.module.values.api.API.addDollars
import su.hubs.hubscore.module.values.api.API.takeHubixes
import su.hubs.hubscore.util.MessageUtils
import su.hubs.hubscore.util.PlayerUtils

class HubsDonate : CoreModule() {

    override fun onEnable(): Boolean {
        PluginUtils.setCommandExecutorAndTabCompleter(DonateCommand(), CartCommand())
        return true
    }

    override fun onDisable() {}

    override fun onReload() {}

    override fun onPlayerJoin(player: Player) {}

    override fun onPlayerLeave(player: Player) {}

    override fun onSchedule(min: Byte) {
        CartManager.makeCycle()
    }

    override fun onCommandExecute(sender: CommandSender, command: Command, label: String, args: Array<String>) = false

    override fun onTabComplete(sender: CommandSender, command: Command, alias: String, args: Array<String>) = null

    companion object {
        fun buyDonate(player: Player, donateKey: String, prevMenu: ChestertonMenu, prevItem: ChestertonItem) {
            val section = PluginUtils.getConfigInCoreFolder("donate").getConfigurationSection(donateKey)
            val price = section?.getInt("price") ?: 999
            val menu = ConfirmMenu(
                    prevItem.material,
                    prevItem.createItemStack(player).itemMeta?.displayName,
                    ValueType.HUBIXES,
                    price
            ) { _, _ ->
                if (!takeHubixes(player, price)) {
                    MessageUtils.sendNotEnoughHubixesMessage(player)
                } else {
                    when (section?.getString("type")) {
                        "TEMP_PERM" -> {
                        }
                        "TEMP_GROUP" -> {
                            Bukkit.getServer().dispatchCommand(Bukkit.getConsoleSender(),
                                    "lp user ${player.name} parent removetemp ${section.getString("group")}"
                            )
                            Bukkit.getServer().dispatchCommand(Bukkit.getConsoleSender(),
                                    "lp user ${player.name} parent addtemp ${section.getString("group")} ${section.getString("long")}"
                            )
                        }
                        "DOLLAR_PACK" -> {
                            addDollars(player, section.getInt("dollars"))
                        }
                        "EXP_PACK", "LVL_PACK", "ITEM_SET" -> {
                            CartManager.addItemToCart(player, donateKey)
                        }
                    }
                    HubsEssentials.sendRawtextMessage("donate_$donateKey", player)
                }
                true
            }
            menu.parentMenu = prevMenu
            PlayerUtils.openMenuToPlayer(player, menu)
        }
    }
}